packages/jest-config/src/normalize.ts

```diff
@@ -142,7 +143,13 @@ const setupPreset = (
       delete require.cache[require.resolve(presetModule)];
     } catch {}
 
-    preset = require(presetModule);
+    if (path.extname(presetModule) === '.mjs') {
```

Comment:
instead of checking file extension, you should just `require` and check if the error thrown is `ERR_REQUIRE_ESM`, similar to https://github.com/facebook/jest/blob/33c3b0718169ffcacce1d2a3ad6b7d77f27deec9/packages/jest-config/src/readConfigFileAndSetRootDir.ts#L34-L63

Reply:
thanks for the hint !